bindata/network/ovn-kubernetes/alert-rules-control-plane.yaml

```diff
@@ -17,7 +17,7 @@ spec:
         message: |
           there is no running ovn-kubernetes master
       expr: |
-        absent(up{job="ovnkube-master",namespace="openshift-ovn-kubernetes"} ==1)
+        absent(up{job="ovnkube-master-metrics",namespace="openshift-ovn-kubernetes"} ==1)
```

Comment:
If the job name is the name of the service, should it be `ovn-kubernetes-master-metrics` instead as per: [https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/ovn-kubernetes/monitor.yaml#L33](https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/ovn-kubernetes/monitor.yaml#L33)

Reply:
The job name is a Kube Service.

Reply:
I think we should target the "ovnkube-db" service though per https://github.com/openshift/cluster-network-operator/pull/742#issuecomment-668134977

Reply:
I tried changing it to ovnkube-db, the alert still fires. I might have to add an additional ServiceMonitor for us to use this service from what I understand (https://coreos.com/blog/the-prometheus-operator.html)

Also I think its ok to leave it as ovnkube-master-metrics since in this instance the metrics pod's endpoint at port 9102 is indirectly connected to master container's 21902, it should be checking for the master pods' endpoint while scraping right ?

Reply:
Also I think the job name could be basically the label defined in the `jobLabel` field of the respective ServiceMonitor (https://coreos.com/operators/prometheus/docs/latest/api.html#servicemonitorspec)

Reply:
> Also I think its ok to leave it as ovnkube-master-metrics since in this instance the metrics pod's endpoint at port 9102 is indirectly connected to master container's 21902, it should be checking for the master pods' endpoint while scraping right ?

this assumption is wrong, it indeed only looks for metrics pods..

Reply:
all right so after some trial and errors, this is what I found: Whatever is the job name provided - in our case this is the `jobLabel` field value in the `ServiceMonitor` object which is `app: ovnkube-master` ; prometheus will scrape the endpoints which is the `instance:portno.` on all the pods which have the same label i.e `app:ovnkube-master` in our case. I think this is what @abhat  was also telling me on slack.

So for our particular alert query `up{job="ovnkube-master",namespace="openshift-ovn-kubernetes"} == 1` this is the output when all is well:



Name | endpoint | instance | job | namespace | pod | prometheus | service | Value
-- | -- | -- | -- | -- | -- | -- | -- | --
   up | metrics | 10.0.0.3:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-hnntq | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1
   up | metrics | 10.0.0.3:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-826kq | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1
   up | metrics | 10.0.0.4:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-frx9r | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1
   up | metrics | 10.0.0.4:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-xh5qb | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1
   up | metrics | 10.0.0.5:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-jppw6 | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1
   up | metrics | 10.0.0.5:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-zz42l | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 1

When I brought down the metrics pods this is what it looks like:
and the alert gets fired.

Name | endpoint | instance | job | namespace | pod | prometheus | service | Value
-- | -- | -- | -- | -- | -- | -- | -- | --
  up | metrics | 10.0.0.3:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-hnntq | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0
  up | metrics | 10.0.0.4:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-xh5qb | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0
  up | metrics | 10.0.0.5:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-jppw6 | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0


When I brought down the master pods this is what it looks like:
and the alert is fired.


Name | endpoint | instance | job | namespace | pod | prometheus | service | Value
-- | -- | -- | -- | -- | -- | -- | -- | --
   up | metrics | 10.0.0.3:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-ghvlw | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0
   up | metrics | 10.0.0.4:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-kmftc | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0
   up | metrics | 10.0.0.5:9102 | ovnkube-master | openshift-ovn-kubernetes | ovnkube-master-metrics-7jxjp | openshift-monitoring/k8s | ovn-kubernetes-master-metrics | 0


Another point to note is that this particular alert `NoRunningOVNKubeMaster` is duplicating what the global alert `TargetDown` does. TargetDown essentially scraps the up metrics for all openshift components. This alert is controlled by the cluster-monitoring-operator. Hence even if we don't have a seperate alert for OVNKube nodes, TargetDown takes care of this when the pod goes down (or used to take care, now it takes care of the ovnkube-node-metrics pods).

Nevertheless we can still continue to have our own alert since its been there from the beginning.